src/net_processing.cpp

```diff
@@ -2866,10 +2898,17 @@ void ProcessMessage(
                 uint32_t nFetchFlags = GetFetchFlags(pfrom);
                 const auto current_time = GetTime<std::chrono::microseconds>();
 
-                for (const CTxIn& txin : tx.vin) {
-                    CInv _inv(MSG_TX | nFetchFlags, txin.prevout.hash);
-                    pfrom.AddInventoryKnown(txin.prevout.hash);
-                    if (!AlreadyHave(_inv, mempool)) RequestTx(State(pfrom.GetId()), _inv.hash, current_time);
+                if (!State(pfrom.GetId())->m_wtxid_relay) {
+                    for (const CTxIn& txin : tx.vin) {
+                        // Here, we only have the txid (and not wtxid) of the
```

Comment:
This will only request parents of orphan transactions when received from non-wtxid peers? What if all peers support wtxid?

Can't we still request them, but with MSG_TX instead of MSG_WTX? That'd require keeping track of wtxidness per request, unfortunately.

Reply:
[Earlier discussion of this](https://github.com/bitcoin/bitcoin/pull/18044#discussion_r400444597) proposed handling this via package relay

Reply:
I think this will be an issue even when there's only a small number of wtxid peers -- in that case we'll delay requesting transactions from non-wtxid peers, so most of the txs (and hence most of the orphans) we receive will be from wtxid peers. I guess I'd still rather see this merged now and fixed later than further delayed though?

Reply:
That seems crazy to me. Do we think that parent fetching of orphans is so unimportant that breaking is not a problem?

I agree with the longer term plan that a better way of fetching unconfirmed parents is needed, but I don't think we can just break it in a PR without even having code for what the solution will look like?

Reply:
Ok, I read back some of the history there. I see the complexity it introduces, but I'm still really uncomfortable with just breaking orphan parent-fetching. I'll instrument my node to see how frequently that actually happens.

Reply:
Hmm, it just doesn't seem like a big deal to me. I think https://github.com/ajtowns/bitcoin/commits/202007-wtxid-orphans should be all that's needed to fix it.

Reply:
@ajtowns That does not look bad at all, indeed. I guess we can move ahead with this and re-instate orphan fetching with a patch like that afterwards.

Reply:
Seems like it would fit in well with the tx relay overhaul, if you stole another bit from m_sequence for the txid vs wtxid flag to me.